src/Lightweight/DataMapper/DataMapper.hpp

```diff
@@ -1142,6 +1170,40 @@ std::vector<Record> DataMapper::Query(std::string_view sqlQueryString, InputPara
     return result;
 }
 
+template <typename FirstRecord, typename SecondRecord, typename... InputParameters>
+    requires DataMapperRecord<FirstRecord> && DataMapperRecord<SecondRecord>
+std::vector<std::tuple<FirstRecord, SecondRecord>> DataMapper::Query(
+    SqlSelectQueryBuilder::ComposedQuery const& selectQuery, InputParameters&&... inputParameters)
+{
+    auto result = std::vector<std::tuple<FirstRecord, SecondRecord>> {};
+
+    _stmt.Prepare(selectQuery.ToSql());
+    _stmt.Execute(std::forward<InputParameters>(inputParameters)...);
+
+    const auto ConfigureFetchAndBind = [this](auto& record) {
+        auto& [recordFirst, recordSecond] = record;
+        BindOutputColumns<FirstRecord, 1>(recordFirst);
+        BindOutputColumns<SecondRecord, Reflection::CountMembers<FirstRecord> + 1>(recordSecond);
+        ConfigureRelationAutoLoading(recordFirst);
+        ConfigureRelationAutoLoading(recordSecond);
+    };
+
+    auto& firstRecord = result.emplace_back();
+    ConfigureFetchAndBind(firstRecord);
+
+    while (_stmt.FetchRow())
+    {
+        auto& record = result.emplace_back();
+        ConfigureFetchAndBind(record);
+    }
```

Comment:
```suggestion
        ConfigureFetchAndBind(result.emplace_back());
```
I am not sure, but I think there were cases where pre-binding did not work. We may need to take care of that. We do that in other places already, we can talk tomorrow (Monday) in person about it. 🤔